lib/src/package_analysis/shapes.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
+
+part 'shapes.g.dart';
+
+/// A Shape for describing a package.
+@sealed
+@JsonSerializable()
+class PackageShape {
+  final List<LibraryShape> libraries;
+
+  /// All the classes defined in the libraries of this package.
+  final Set<ClassShape> classes;
+
+  PackageShape({required this.libraries, required this.classes});
+
+  factory PackageShape.fromJson(Map<String, dynamic> json) =>
+      _$PackageShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$PackageShapeToJson(this);
+}
+
+/// A Shape for describing a library, which may consist of multiple parts.
+@sealed
+@JsonSerializable()
+class LibraryShape {
+  /// Unique identifier for the library represented by this [LibraryShape]
+  /// within the given [PackageShape]. This uses the syntax beginning with
+  /// `package:` which would usually be used for importing this library.
+  ///
+  /// Example:
+  /// ```
+  /// package:my_package/my_library.dart
+  /// ```
+  final String uri;
+
+  final List<PropertyShape> getters;
+  final List<PropertyShape> setters;
+
+  /// The top-level functions exported in this library.
+  final List<ExecutableShape> functions;
```

Comment:
I think we should consider creating `id`s for these as well... because of importing two conflicting definitions work, we probably need to consider it a breaking change to re-defined a previous re-exported symbol.